x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvSort.java

```diff
@@ -144,10 +149,24 @@ public boolean foldable() {
     public EvalOperator.ExpressionEvaluator.Factory toEvaluator(
         Function<Expression, EvalOperator.ExpressionEvaluator.Factory> toEvaluator
     ) {
-        Expression nonNullOrder = order == null ? ASC : order;
-        boolean ordering = nonNullOrder.foldable() && ((BytesRef) nonNullOrder.fold()).utf8ToString().equalsIgnoreCase("DESC")
-            ? false
-            : true;
+        boolean ordering = true;
+        if (isValidOrder() == false) {
+            throw new IllegalArgumentException(
+                "Invalid order value in ["
```

Comment:
Did you mean to use `invalidOrderError` here?

Reply:
> Did you mean to use `invalidOrderError` here?

After discussing with @astefan , we decide to use `IllegalArgumentException` instead of `InvalidArgumentException`, so the `invalidOrderError` cannot be used directly here, as it takes only one string.

Reply:
You can use `LoggerMessageFormat.format(INVALID_ORDER_ERROR, sourceText(), ASC.value(), DESC.value(), ((BytesRef) order.fold()).utf8ToString())` to build one string, same as `InvalidArgumentException` c'tor and `Failure.fail()` does.